jmh.gradle

```diff
@@ -21,10 +21,15 @@ if (jdkVersion != '8' && jdkVersion != '11' && jdkVersion != '17') {
   throw new GradleException("The JMH benchamrks must be run with JDK 8 or JDK 11 or JDK 17")
 }
 
+def flinkVersions = (System.getProperty("flinkVersions") != null ? System.getProperty("flinkVersions") : System.getProperty("defaultFlinkVersions")).split(",")
 def sparkVersions = (System.getProperty("sparkVersions") != null ? System.getProperty("sparkVersions") : System.getProperty("defaultSparkVersions")).split(",")
 def scalaVersion = System.getProperty("scalaVersion") != null ? System.getProperty("scalaVersion") : System.getProperty("defaultScalaVersion")
 def jmhProjects = [project(":iceberg-core")]
 
+if (flinkVersions.contains("1.17")) {
```

Comment:
nit: Could we move the whole patch to v1.18, and add it there first?

Reply:
I think we can continue the initial PR on 1.17. will port to 1.16 and 1.18. don't see much difference on which version we work on first. this is incomplete feature anyway.